java-checks/src/main/java/org/sonar/java/checks/ReturnInFinallyCheck.java

```diff
@@ -23,21 +23,27 @@
 import org.sonar.plugins.java.api.JavaFileScanner;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
 import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.BlockTree;
 import org.sonar.plugins.java.api.tree.BreakStatementTree;
 import org.sonar.plugins.java.api.tree.ContinueStatementTree;
+import org.sonar.plugins.java.api.tree.DoWhileStatementTree;
+import org.sonar.plugins.java.api.tree.ForEachStatement;
+import org.sonar.plugins.java.api.tree.ForStatementTree;
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.ReturnStatementTree;
+import org.sonar.plugins.java.api.tree.SwitchStatementTree;
 import org.sonar.plugins.java.api.tree.ThrowStatementTree;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.TryStatementTree;
+import org.sonar.plugins.java.api.tree.WhileStatementTree;
 
 import java.util.Deque;
 import java.util.LinkedList;
 
 @Rule(key = "S1143")
 public class ReturnInFinallyCheck extends BaseTreeVisitor implements JavaFileScanner {
 
-  private final Deque<Boolean> isInFinally = new LinkedList<>();
+  private final Deque<Tree.Kind> isInFinally = new LinkedList<>();
```

Comment:
I would change the variable name as it's counterintuitive now. `nestedTreeKind`?